src/middleware/error-handler.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+
+// Define a custom error type if needed
+interface CustomError extends Error {
+  statusCode?: number;
+  code?: number;
+  errors?: Record<string, { message: string }>;
+  keyValue?: Record<string, any>;
+  value?: string;
+}
+
+export const errorHandlerMiddleware = (
```

Comment:
I would personally advocate for, wherever these ValidationErrors or CastErrors are returned, instead we should return the exact message and code we want if possible. If these errors are produced by code that is not written by us (i.e. MongoDB is producing these errors), let's find a way to handle that in the Mongoose code that we write.

If I have an error that is returned randomly somewhere in my application but gets special treatment in this middleware, it will be VERY difficult to understand where the original error is when I am trying to investigate why a particular error has produced by this code.